include/mqtt/broker/security.hpp

```diff
@@ -0,0 +1,492 @@
+// Copyright Wouter van Kleunen 2021
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(MQTT_BROKER_SECURITY_HPP)
+#define MQTT_BROKER_SECURITY_HPP
+
+#include <string>
+
+#include <mqtt/broker/broker_namespace.hpp>
+#include <mqtt/broker/subscription_map.hpp>
+#include <mqtt/optional.hpp>
+
+#include <map>
+#include <set>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/optional.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/algorithm/string.hpp>
+
+#if MQTT_USE_TLS
+#include <openssl/evp.h>
+#endif
+
+MQTT_BROKER_NS_BEGIN
+
+struct security
+{
+
+    struct authentication
+    {
+        enum class method {
+            password,
+            client_cert,
+            anonymous
+        };
+
+        method method_;
+        optional<std::string> password;
+
+        authentication(method method_ = method::password, optional<std::string> const &password = optional<std::string>())
+            : method_(method_), password(password)
+        { }
+
+        std::vector<std::string> groups;
+    };
+
+
+    struct authorization
+    {        
+        enum class type {
+            deny, allow
+        };
+
+        std::vector<std::string> topic_tokens;
+
+        type sub_type;
+        std::set<std::string> sub;
+
+        type pub_type;
+        std::set<std::string> pub;
+
+        authorization()
+            : sub_type(type::deny), pub_type(type::deny)
+        { }
+    };
+
+    struct group
+    {
+        std::string name;
+        std::vector<std::string> members;
+    };
+
+
+    optional<std::string> login_anonymous() {
+        return anonymous;
+    }
+
+    template<typename T>
+    static inline std::string to_hex(T const start, T const end)
+    {
```

Comment:
```suggestion
    static std::string to_hex(T const start, T const end) {
```

remove inline and place `{` in the same line.
Please fix other same cases.